### PR TITLE
secret-54 major changes to database and fixes

### PR DESCRIPTION
**Pull request by ZleMyzteX**
* changed / fixed database schema
* added basic architecture tests
* update dependencies
* added lombok

### CODE REVIEW

**Review by leakonvalinka — CHANGES_REQUESTED** [review]
vl sollte man die controller rausnehmen? die sollen ja dann die andern schreiben
bevor mans auf main schallert

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/UserRepository.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    @Transactional
+    public User readById(UUID uuid) {
+        List<User> registered = this.dslContext.selectFrom(AppUser.APP_USER)
+                .where(AppUser.APP_USER.UUID.equal(uuid))
+                .fetchInto(User.class);
+        return registered.stream().findFirst().orElse(null);
+    }
+
+    @Transactional
+    public User update(User newUser) {
```
Comment (leakonvalinka) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/UserRepository.java:64]: das setzt nur values von registered users, also guest update fehlt noch (falls das möglich sein soll? kann man überlegen ob man das braucht)

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/endpoint/controller/UserController.java [hunk 2]
```diff
@@ -39,6 +39,7 @@ public UserController(UserRepository userRepository, FirebaseAuth firebaseAuth)
   @PostMapping()
   @ResponseStatus(HttpStatus.CREATED)
   public User create(@RequestBody User newUser) {
+    System.out.println("creating new user: " + newUser);
```
Comment (ZleMyzteX) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/endpoint/controller/UserController.java]: if we want to log stuff (temporarily, even) we should use a logger
-> here I'd suggest to simply use the @log4j annotation & put a log.debug("...", newUser) into this.

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/GroupRepository.java [hunk 3]
```diff
@@ -44,26 +45,21 @@ public List<Group> readAll() {
 
   @Transactional
   public Group readById(UUID uuid) {
```
Comment (ZleMyzteX) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/GroupRepository.java]: we should change the name back to UUID, readById is wrong at the moment

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/GroupRepository.java [hunk 4]
```diff
@@ -44,26 +45,21 @@ public List<Group> readAll() {
...
-            .returning()
-            .fetchOne();
-
-    return this.toGroup(groupRec);
+    return this.dslContext
+        .update(AppGroup.APP_GROUP)
+        .set(this.toRecord(newGroup))
+        .where(AppGroup.APP_GROUP.UUID.equal(newGroup.getUuid()))
+        .returning()
+        .fetchOneInto(Group.class);
```
Comment (ZleMyzteX) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/GroupRepository.java:62]: nice

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/ParticipantRepository.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    return this.dslContext
+        .insertInto(AppParticipant.APP_PARTICIPANT)
+        .set(this.toRecord(newParticipant))
+        .returning()
+        .fetchOneInto(Participant.class);
+  }
+
+  @Transactional
+  public List<Participant> readAllForUser(UUID userUuid) {
```
Comment (ZleMyzteX) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/ParticipantRepository.java]: I'd maybe change the name here; readAllForUser is a little bit confusing?

readAllByUser maybe? 🤔

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/ParticipantRepository.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    AppParticipantRecord participantRecord = new AppParticipantRecord();
+
+    participantRecord.setCreated(participant.getCreated());
+    participantRecord.setLastModified(participant.getLastModified());
+    participantRecord.setUserUuid(participant.getUserUuid());
+    participantRecord.setGroupUuid(participant.getGroupUuid());
+    participantRecord.setExcluded(participant.getExcludedUser());
+    participantRecord.setAvoid(participant.getAvoidUser());
+    participantRecord.setInterests(participant.getInterests());
+    // participantRecord.setBengerlUuid(participant.getBengerlUuid());
```
Comment (ZleMyzteX) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/ParticipantRepository.java:87]: why is this commented out? 

and maybe we should really switch to mapstruct or something similar

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/UserRepository.java [hunk 7]
```diff
@@ -25,75 +25,56 @@ public UserRepository(DSLContext dslContext) {
...
-
-    AppUserRecord registeredUser =
-        this.dslContext
-            .insertInto(AppUser.APP_USER)
-            .set(this.toRecord(newUser))
-            .returning()
-            .fetchOne();
-
-    return this.toUser(registeredUser);
+    LocalDateTime now = LocalDateTime.now();
```
Comment (ZleMyzteX) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/UserRepository.java]: we should maybe put this into a LocalDateTimeUtil class or something like that to share that logic

File: SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/UserRepository.java [hunk 8]
```diff
@@ -25,75 +25,56 @@ public UserRepository(DSLContext dslContext) {
...
         this.dslContext
             .selectFrom(AppUser.APP_USER)
             .where(AppUser.APP_USER.FIREBASE_UID.equal(firebaseUID))
-            .fetchInto(User.class);
-
-    // TODO: validate if more than one match (gg)
-
-    return users.stream().findFirst().orElse(null);
+            .fetchOneInto(User.class);
+    return user;
```
Comment (ZleMyzteX) [inline SecretSalty/src/main/java/com/secretsalty/secretsalty/repository/UserRepository.java]: can simply return, no assignment needed here

**Pull request merged**